src/renderer/dx/DxFontInfo.cpp

```diff
@@ -335,48 +335,14 @@ void DxFontInfo::SetFromEngine(const std::wstring_view familyName,
 // - dwriteFactory - The DWrite factory to use
 // Return Value:
 // - DirectWrite font collection. May be null if one cannot be created.
-[[nodiscard]] IDWriteFontCollection* DxFontInfo::_NearbyCollection(gsl::not_null<IDWriteFactory1*> dwriteFactory)
+[[nodiscard]] IDWriteFontCollection* DxFontInfo::_NearbyCollection()
 {
-    if (_nearbyCollection)
-    {
-        return _nearbyCollection.Get();
-    }
-
-    // The convenience interfaces for loading fonts from files
-    // are only available on Windows 10+.
-    ::Microsoft::WRL::ComPtr<IDWriteFactory6> factory6;
-    if (FAILED(dwriteFactory->QueryInterface<IDWriteFactory6>(&factory6)))
+    if (!_nearbyCollection)
     {
-        return nullptr;
+        _nearbyCollection = FontCache::GetCached();
     }
 
-    ::Microsoft::WRL::ComPtr<IDWriteFontCollection1> systemFontCollection;
-    THROW_IF_FAILED(factory6->GetSystemFontCollection(false, &systemFontCollection, 0));
-
-    ::Microsoft::WRL::ComPtr<IDWriteFontSet> systemFontSet;
-    THROW_IF_FAILED(systemFontCollection->GetFontSet(&systemFontSet));
-
-    ::Microsoft::WRL::ComPtr<IDWriteFontSetBuilder2> fontSetBuilder2;
-    THROW_IF_FAILED(factory6->CreateFontSetBuilder(&fontSetBuilder2));
-
-    THROW_IF_FAILED(fontSetBuilder2->AddFontSet(systemFontSet.Get()));
-
-    // Magic static so we only attempt to grovel the hard disk once no matter how many instances
-    // of the font collection itself we require.
-    static const auto knownPaths = s_GetNearbyFonts();
-    for (auto& p : knownPaths)
-    {
-        fontSetBuilder2->AddFontFile(p.c_str());
-    }
-
-    ::Microsoft::WRL::ComPtr<IDWriteFontSet> fontSet;
-    THROW_IF_FAILED(fontSetBuilder2->CreateFontSet(&fontSet));
-
-    ::Microsoft::WRL::ComPtr<IDWriteFontCollection1> fontCollection;
-    THROW_IF_FAILED(factory6->CreateFontCollectionFromFontSet(fontSet.Get(), &fontCollection));
-
-    _nearbyCollection = fontCollection;
-    return _nearbyCollection.Get();
+    return _nearbyCollection.get();

```

Comment:
since FontCache is managing the cache... do we need to keep `_nearbyCollection` at all? Won't it result in torn state when the cache is flushed, but existing dwrite engines use the OLD cache?